src/transformers/models/dbrx/modeling_dbrx.py

```diff
@@ -0,0 +1,2132 @@
+                                 fill_value=min_dtype,
+                                 dtype=dtype,
+                                 device=device)
+        if sequence_length != 1:
+            causal_mask = torch.triu(causal_mask, diagonal=1)
+        causal_mask *= torch.arange(
+            target_length, device=device) > cache_position.reshape(-1, 1)
+        causal_mask = causal_mask[None,
+                                  None, :, :].expand(input_tensor.shape[0], 1,
+                                                     -1, -1)
+        if attention_mask is not None:
+            causal_mask = causal_mask.clone(
+            )  # copy to contiguous memory for in-place edit
+            if attention_mask.dim() == 2:
+                mask_length = attention_mask.shape[-1]
+                padding_mask = causal_mask[..., :mask_length].eq(
+                    0.0) * attention_mask[:, None, None, :].eq(0.0)
+                causal_mask[..., :mask_length] = causal_mask[
+                    ..., :mask_length].masked_fill(padding_mask, min_dtype)
+            elif attention_mask.dim() == 4:
+                # backwards compatibility: we allow passing a 4D attention mask shorter than the input length with
+                # cache. In that case, the 4D attention mask attends to the newest tokens only.
+                if attention_mask.shape[
+                        -2] < cache_position[0] + sequence_length:
+                    offset = cache_position[0]
+                else:
+                    offset = 0
+                mask_shape = attention_mask.shape
+                mask_slice = (attention_mask.eq(0.0)).to(
+                    dtype=dtype) * min_dtype
+                causal_mask[:mask_shape[0], :mask_shape[1],
+                            offset:mask_shape[2] +
+                            offset, :mask_shape[3]] = mask_slice
+
+        if (self.config._attn_implementation == 'sdpa' and
+                attention_mask is not None and
+                attention_mask.device.type == 'cuda'):
+            # TODO: For dynamo, rather use a check on fullgraph=True once this is possible (https://github.com/pytorch/pytorch/pull/120400).
+            is_tracing = (
+                torch.jit.is_tracing() or
+                isinstance(input_tensor, torch.fx.Proxy) or  # type: ignore
+                (hasattr(torch, '_dynamo') and torch._dynamo.is_compiling()))
+            if not is_tracing and torch.any(attention_mask != 1):
+                # Attend to all tokens in fully masked rows in the causal_mask, for example the relevant first rows when
+                # using left padding. This is required by F.scaled_dot_product_attention memory-efficient attention path.
+                # Details: https://github.com/pytorch/pytorch/issues/110213
+                causal_mask = AttentionMaskConverter._unmask_unattended(
+                    causal_mask, min_dtype)
+
+        return causal_mask
+
+
```

Comment:
copied from should be used

Reply:
I think there are changes from the Mixtral config that make `copied from` not work here